### PR TITLE
Ensure required builder.conf variables are present

### LINKED ISSUE #63: Mixer fails with permission denied when version path is not set
Steps to reproduce: 
- Start with empty workspace
- Set builder conf to not have `VERSIONS_PATH` or misspelled, i.e `VERSIONS_DIR`
- Run:
```
$ mixer init-mix --clear-version 20010 --mix-version 10
Mixer 3.2.1
***Error: open /.clearurl: permission denied
```

Works when run with sudo, but this command should not need sudo privileges.

### PR DESCRIPTION
**Pull request by mdhorn**
Check for required variables in the builder.conf when reading.

Fixes #63

Signed-off-by: Mark D Horn <mark.d.horn@intel.com>

### CODE REVIEW

**Comment by mdhorn** [comment]
This ensures that "required" values are present with a value in the builder.conf.
@tmarcu please ensure I have the correct list of required variables.

This is prevent the "permission" issue reported, but the user may still see an error of the parent directory does not exist; i.e.:
```
mixer init-mix --clear-version 20010 --mix-version 10
Mixer 3.2.1
***Error: open /home/clr/mix/.clearurl: no such file or directory
```

File: builder/builder.go [hunk 1]
```diff
@@ -113,19 +113,20 @@ func (b *Builder) ReadBuilderConf() {
...
-		{`^FORMAT\s*=\s*`, &b.Format},
-		{`^MIXVER\s*=\s*`, &b.Mixver},
-		{`^REPODIR\s*=\s*`, &b.Repodir},
-		{`^RPMDIR\s*=\s*`, &b.Rpmdir},
-		{`^SERVER_STATE_DIR\s*=\s*`, &b.Statedir},
-		{`^VERSIONS_PATH\s*=\s*`, &b.Versiondir},
-		{`^YUM_CONF\s*=\s*`, &b.Yumconf},
+		{`^BUNDLE_DIR\s*=\s*`, &b.Bundledir, true},
+		{`^CERT\s*=\s*`, &b.Cert, true},
+		{`^CLEARVER\s*=\s*`, &b.Clearver, false},
```
Comment (tmarcu) [inline builder/builder.go]: Are you guys using this variable in the conf? Or the MIXVER one?

Comment (tmarcu) [inline builder/builder.go]: Are you guys using this variable in the conf? Or the MIXVER one?

Comment (mdhorn) [reply builder/builder.go]: We aren't using them directly in our embargoed tooling.
The ones I marked as 'required' were based on my guess and what the mixer code appears to rely upon.

Comment (tmarcu) [reply builder/builder.go]: Ok, I think those aren't use anywhere at this point, so I'll remove them in a different patch. The rest look correct to me though!

**Comment by cmarcelo** [comment]
Apparently this needs a rebase.

**Comment by mdhorn** [comment]
@cmarcelo I've rebased and repushed the updated commit.
@tmarcu Can we merge this soon since it was approved back on January 4th? I don't have any merge rights.

**Pull request merged**